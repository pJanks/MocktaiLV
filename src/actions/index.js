export const addUser = (user) => ({
  type: 'ADD_USER',
  user
})

export const addDrinks = (drinks) => ({
  type: 'LOAD_NONALCOHOLIC_DRINKS',
  drinks
})

export const addFavoriteDrink = (favoriteDrink) => ({
  type: 'ADD_FAVORITE_DRINK',
  favoriteDrink
})

export const getDrink = (drink) => ({
  type: 'GET_DRINK',
  drink
})
