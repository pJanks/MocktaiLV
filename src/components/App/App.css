* {
  box-sizing: border-box;
  color: #FFF;
  margin: 0;
  padding: 0;
}

.App {
  background-color: #7692FF;
  height: 100vh;
  width: 100%;
  text-align: center;
}

img {
  height: 300px;
  width: 300px;
}
