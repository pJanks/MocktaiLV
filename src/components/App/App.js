import React, { Component } from 'react';
import { Route, Switch } from 'react-router-dom';
import { connect } from 'react-redux';
import Login from '../Login/Login'
import AllMocktails from '../AllMocktails/AllMocktails'
import DrinkDetails from '../DrinkDetails/DrinkDetails'
import SignUp from '../SignUp/SignUp'
import './App.css';

export class App extends Component {
  constructor() {
    super()
  }

  render() {
    return (
      <main className="App">
        <header>
          <h1>MocktaiLV</h1>
        </header>
        <Switch>
          <Route exact path='/' component={Login} />
          <Route exact path='/AllMocktails' component={AllMocktails} />
          <Route exact path={`/DrinkDetails${this.props.selectedDrink.idDrink}`} render={() => <DrinkDetails drink={this.props.selectedDrink} />} />
          <Route exact path='/SignUp' component={SignUp} />
          <BadPathPage>
        </Switch>
      </main>
    );
  }
}

export const mapStateToProps = (state) => ({
  selectedDrink: state.selectedDrink,
})

export default connect(mapStateToProps)(App);
