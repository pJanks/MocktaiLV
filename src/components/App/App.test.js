import React from 'react';
import { App, mapStateToProps } from './App';
import { addDrinks } from '../../actions'
import { shallow } from 'enzyme'

describe('App', () => {
  let wrapper, instance;

  beforeEach(() => {
    wrapper = shallow(<App selectedDrink={ {
      strDrink: "Afterglow",
      strDrinkThumb: "https://www.thecocktaildb.com/images/media/drink/vuquyv1468876052.jpg",
      idDrink: "12560"
    } } />)
    instance = wrapper.instance()
  })

  it('should match a snapshot as expected', () => {  
    expect(wrapper).toMatchSnapshot()
  })

  describe('mockStateToProps', () => {

    it('should return a selectedDrink object', () => {
      const mockState = { selectedDrink: {
        strDrink: "Afterglow",
        strDrinkThumb: "https://www.thecocktaildb.com/images/media/drink/vuquyv1468876052.jpg",
        idDrink: "12560"
      } }
      const expected = {
        strDrink: "Afterglow",
        strDrinkThumb: "https://www.thecocktaildb.com/images/media/drink/vuquyv1468876052.jpg",
        idDrink: "12560"
      }
      const mappedProps = mapStateToProps( mockState );
      expect(mappedProps.selectedDrink).toEqual(expected)
    })
  })
})
