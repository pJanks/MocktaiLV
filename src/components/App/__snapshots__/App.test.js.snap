// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match a snapshot as expected 1`] = `
<main
  className="App"
>
  <header>
    <h1>
      MocktaiLV
    </h1>
  </header>
  <Switch>
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(Login)",
          "type": [Function],
        }
      }
      exact={true}
      path="/"
    />
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(AllMocktails)",
          "type": [Function],
        }
      }
      exact={true}
      path="/AllMocktails"
    />
    <Route
      exact={true}
      path="/DrinkDetails12560"
      render={[Function]}
    />
    <Route
      component={[Function]}
      exact={true}
      path="/SignUp"
    />
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(OffTheWagon)",
          "type": [Function],
        }
      }
      exact={true}
      path="/OffTheWagon"
    />
    <Route
      exact={true}
      path="/Favorites"
      render={[Function]}
    />
    <BadPathPage />
  </Switch>
  <Nav />
</main>
`;
